rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -0,0 +1,178 @@
+    /*
+     * Spinning up the separate JVMs for bookKeeper and Local Data Transfer Servers
+     * */
+    String[] bookKeeperStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, bookKeeperClass, setDataBlockSize, setCacheMaxDisks, setCacheDirectory};
+    String[] localDataTransferStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, localDataTransferServerClass, setDataBlockSize, setCacheMaxDisks, setCacheDirectory};
+
+    ProcessBuilder pJVMBuilder = new ProcessBuilder();
+    pJVMBuilder.redirectErrorStream(true);
+    pJVMBuilder.inheritIO();
+    pJVMBuilder.command(bookKeeperStartCmd);
+    bookKeeperJvm = pJVMBuilder.start();
+    pJVMBuilder.command(localDataTransferStartCmd);
+    localDataTransferJvm = pJVMBuilder.start();
+
+    log.info("Test Directory : " + testDirectory);
+    Files.createDirectories(Paths.get(testDirectoryPrefix + "dir0/fcache/"));
+    Thread.sleep(3000);
+  }
+
+  @AfterClass
+  public static void tearDownClass() throws IOException
+  {
+    /* ****
+      Destroying bookKeeper and localDataTransfer JVMs
+    **** */
+    bookKeeperJvm.destroy();
+    localDataTransferJvm.destroy();
+
+    log.info("Deleting files in " + testDirectory);
+    Files.walkFileTree(Paths.get(testDirectory), new DeleteFileVisitor());
+    Files.deleteIfExists(Paths.get(testDirectory));
+  }
+
+  @BeforeMethod
+  public static void setup() throws IOException, InterruptedException, URISyntaxException
+  {
+    DataGen.populateFile(backendFileName);
+    log.info("BackendPath: " + backendPath);
+
+    CacheConfig.setIsStrictMode(conf, true);
+    CacheConfig.setCacheDataDirPrefix(conf, testDirectoryPrefix + "dir");
+    CacheConfig.setMaxDisks(conf, 1);
+    CacheConfig.setBlockSize(conf, 200);
+  }
+
+  @AfterMethod
+  public static void cleanup()
+  {
+    File file = new File(backendFileName);
+    file.delete();
+
+    File mdFile = new File(CacheUtil.getMetadataFilePath(backendPath.toString(), conf));
+    mdFile.delete();
+
+    File localFile = new File(CacheUtil.getLocalPath(backendPath.toString(), conf));
+    localFile.delete();
+  }
+
+  @Test(enabled = false)
+  public void testJVMCommunication() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+    File file = new File(backendFileName);
+    List<BlockLocation> result;
+    int lastBlock = 4;
+    int readSize = 1000;
+
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    result = client.getCacheStatus("file:///" + backendFileName, file.length(), file.lastModified(), 0, lastBlock, 3);
+    assertTrue(result.get(0).getLocation() == Location.LOCAL, "File already cached, before readData call");
+    log.info(" Value of Result : " + result);
+    log.info("Downloading file from path : " + file.toString());
+    boolean dataDownloaded = client.readData("file:///" + backendFileName, 0, readSize, file.length(), file.lastModified(), 3);
```

Comment:
Use `backendPath` here instead of `"file:///" + backendFileName`

Reply:
Addressed